airflow/providers/hashicorp/secrets/vault.py

```diff
@@ -188,12 +204,28 @@ def get_conn_uri(self, conn_id: str) -> Optional[str]:
         :rtype: str
         :return: The connection uri retrieved from the secret
         """
-        if self.connections_path is None:
+        response = self.get_response(conn_id)
+
+        return response.get("conn_uri") if response else None
+
+    def get_connection(self, conn_id: str) -> Optional[Connection]:
+        """
+        Get connection from Vault as secret. Prioritize conn_uri if exists,
+        if not fall back to normal Connection creation.
+
+        :type conn_id: str
+        :rtype: Connection
+        :return: A Connection object constructed from Vault data
+        """
+        response = self.get_response(conn_id)
+        if response is None:
             return None
-        else:
-            secret_path = self.build_path(self.connections_path, conn_id)
-            response = self.vault_client.get_secret(secret_path=secret_path)
-            return response.get("conn_uri") if response else None
+
+        uri = response.get("conn_uri")
+        if uri:
+            return uri
```

Comment:
This won't return a connection object. I think this should be
```python
if uri:
    return Connection(conn_id, uri=uri)

return Connection(conn_id, **response) 
# or _create_connection, but constructor handles this logic pretty well now without secrets backend
```

Reply:
@ashb Based on the changes we made to the constructor for `Connection`, I don't see any reason not to use it here too.

That said, note that this function won't persist the connection to the DB (but neither would `_create_connection`, so no difference really).

Reply:
Wait so what are the changes I should make?

Reply:
I was raising two suggestions:
- the Connection constructor can replace the `_create_connection` function for this use case
- in one case, this function returns a Connection object; in another it returns a uri. I think it should return a Connection both times?